### PR TITLE
UI: Add toggle hotkeys for filters

### PR DESCRIPTION
**Pull request by cg2121**
This adds a hotkey to show and hide scenes.

### CODE REVIEW

**Comment by cg2121** [comment]
I closed #1133 and combined it with this one.

File: UI/window-basic-main.cpp [hunk 1]
```diff
@@ -2065,6 +2065,45 @@ void OBSBasic::CreateFiltersWindow(obs_source_t *source)
...
+
+static bool HideSource(void *data, obs_hotkey_pair_id id, obs_hotkey_t *key,
+		bool pressed)
+{
+	UNUSED_PARAMETER(id);
+	UNUSED_PARAMETER(key);
+
+	obs_source_t *source = static_cast<obs_source_t*>(data);
+
+	if (!pressed || !obs_source_enabled(source) || !source)
```
Comment (RytoEX) [inline UI/window-basic-main.cpp]: Is the null check (`!source`) required here?  Doesn't `obs_source_enabled(source)` already do this check?  See [libobs/obs-source.c#L3543](https://github.com/obsproject/obs-studio/blob/db06a1c12acb50422d61055fc7053e95048d6bb6/libobs/obs-source.c#L3543) and [libobs/obs-internal.h#L60](https://github.com/obsproject/obs-studio/blob/57121478acccd97b6c3cd909e0e8cbf7d7dda05a/libobs/obs-internal.h#L60) (lines 60-75).  If `source` would be evaluated as false, then `obs_object_valid` would return false, and then `obs_source_enabled` would return false, so `!obs_source_enabled` would be true if `!source` is true.  Similarly, if the source is valid and enabled, then `!obs_source_enabled` would be false and `!source` would also be false.  It seems to me that the `!source` condition would never be evaluated when `!obs_source_enabled` is true, and whenever it is evaluated, `!source` will always be false.

I realize that removing `!source` here may make it seem unbalanced compared to the one in `ShowSource` above.  Something to consider.

**Comment by jp9000** [comment]
Hmm, I'm not sure how I feel about this yet.  I need to go over it a bit more.  I'm not sure if I like it fortunately because there's no real need to perform a hide operation on a scene at the source level (rather than a hide at the scene item level).  Same with filters.

However the bigger problem is hotkeys for filters -- filters have an inherent display issue in the hotkey section where I'm not sure if I like the idea of showing filters there.  The hotkeys section would get so incredibly cluttered.  I know there are some cases where hotkeys on filters may be useful, but I'm not sure if they're really applicable to all filters.  The second display issue is making sure to show that the filter is associated with a specific source in the hotkeys section.  The filters would show up as though they're regular sources, when they're more just child filters of a specific source typically.

So when people start talking about filter hotkeys, I just think back to that and can't help but think "maybe there's a better approach", or "maybe the hotkeys section needs more work first".

**Comment by RytoEX** [comment]
Something I keep mulling over for settings is being able to "search" for a setting (or hotkey).  So if you wanted to find a specific setting or hotkey, you could type into a search box and it would either allow you to go directly to that setting/hotkey or would display only matching settings/hotkeys.  If you need examples of what I'm talking about, Chrome, Opera, and Firefox all have a settings UI/UX like this.  I have no idea how much work it would be to implement something like that though.  If that sounds like something worth discussing, we should probably discuss it separate from this PR.

**Comment by jp9000** [comment]
Hm, yea.  That could be an idea to solve the problem, although it doesn't really address the second display issue.  Filters are more contextual to the specific source they're applied to, and I can't help but feel like it needs to show that in the hotkeys section so the user realizes that a bit easier.  But I mean the problem with the hotkeys section is that there just tend to be so many hotkeys as it is, so it makes the hotkeys section feel almost intimidating.

Honestly, the entire settings window kind of needs some reworking all together; I eventually plan to do it.  If/when I get to it, then yea we should have a discussion about how to approach this particular problem.

**Comment by cg2121** [comment]
After thinking about it for a while, visibility toggle for scenes doesn't quite make sense. Now this PR is just for filters.

**Comment by cg2121** [comment]
I will close this until a better UI can be made for the hotkeys.

**Pull request closed without merging**